c/src/point_one/polaris/polaris.c

```diff
@@ -609,7 +669,18 @@ static int OpenSocket(PolarisContext_t* context, const char* endpoint_url,
 
   // Perform SSL handhshake.
   if (SSL_connect(context->ssl) == -1) {
-    P1_Print("SSL handshake failed to %s:%d.\n", endpoint_url, endpoint_port);
+    P1_Print("SSL handshake failed to tcp://%s:%d.\n", endpoint_url,
+             endpoint_port);
+    CloseSocket(context, 1);
+    return POLARIS_ERROR;
+  }
+
+  const SSL_CIPHER* c = SSL_get_current_cipher(context->ssl);
+  if (c == NULL) {
+    P1_Print(
+        "No encryption cipher enabled - connection failed. Are you sure "
```

Comment:
This error is misleading since it sounds like the client doesn't have the cipher enabled, where in fact this is a server issue. Consider "Server failed to negotiate encryption cipher. Verify this port supports TLS 1.2 or better"

Reply:
Yeah, that sounds better. Will change.